src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,157 @@
+import frc.robot.RobotMap;
+
+public class Intake extends SubsystemBase {
+
+    private final CANSparkMax pivotMotor = new CANSparkMax(RobotMap.INTAKE_PIVOT_MOTOR_CAN_ID, MotorType.kBrushless);
+    private final RelativeEncoder pivotEncoder = pivotMotor.getEncoder();
+    private final AbsoluteEncoder pivotAbsoluteEncoder = pivotMotor.getAbsoluteEncoder(Type.kDutyCycle);
+    private final TalonFX intakeTalonFront = new TalonFX(RobotMap.INTAKING_FRONT_MOTOR_CAN_ID);;
+    private final TalonFX intakeTalonBack = new TalonFX(RobotMap.INTAKING_BACK_MOTOR_CAN_ID);
+    private final Follower intakeTalonBackFollower = new Follower(intakeTalonFront.getDeviceID(), false);
+    private ProfiledPIDController pivotController;
+
+    private Optional<Double> lastControlledTime;
+
+    public enum IntakePosition {
+        DEPLOYED,
+        STOWED,
+        CLEAR_OF_CONVEYOR
+    }
+
+    private TreeMap<IntakePosition, Double> intakePositionMap;
+    private IntakePosition latestPosition = IntakePosition.STOWED;
+    private IntakePosition desiredPosition = IntakePosition.STOWED;
+
+    public Intake() {
+        initPivotMotor();
+        initIntakeTalons();
+        intakePositionMap = new TreeMap<IntakePosition, Double>();
+        intakePositionMap.put(IntakePosition.DEPLOYED, IntakeCal.INTAKE_DEPLOYED_POSITION_DEGREES);
+        intakePositionMap.put(IntakePosition.STOWED, IntakeCal.INTAKE_STOWED_POSITION_DEGREES);
+        intakePositionMap.put(IntakePosition.CLEAR_OF_CONVEYOR, IntakeCal.INTAKE_SAFE_POSITION_DEGREES);
+        lastControlledTime = Optional.of(Timer.getFPGATimestamp());
+    }
+
+    public void initPivotMotor() {
+        pivotController = new ProfiledPIDController(IntakeCal.INTAKE_PIVOT_P,
+            IntakeCal.INTAKE_PIVOT_I, IntakeCal.INTAKE_PIVOT_D, new TrapezoidProfile.Constraints(
+                    IntakeCal.PIVOT_MAX_VELOCITY_DEG_PER_SECOND,
+                    IntakeCal.PIVOT_MAX_ACCELERATION_DEG_PER_SECOND_SQUARED));
+        // inverted
+        // encoder gear ratio
+        // current limit
+    }
+
+    public void initIntakeTalons() {
+        TalonFXConfigurator cfgFront = intakeTalonFront.getConfigurator();
+        TalonFXConfigurator cfgBack = intakeTalonFront.getConfigurator();
+
+        TalonFXConfiguration toApply = new TalonFXConfiguration();
+        toApply.CurrentLimits.SupplyCurrentLimit = IntakeConstants.INTAKING_TALONS_CURRENT_LIMIT_AMPS;
+        toApply.CurrentLimits.SupplyCurrentLimitEnable = true;
+        toApply.MotorOutput.NeutralMode = NeutralModeValue.Coast;
+
+        cfgFront.apply(toApply);
+        cfgBack.apply(toApply);
+
+        intakeTalonBack.setControl(intakeTalonBackFollower);
+    }
+
+    /** @return the last controlled time by the timer, if the lastControlledTime is empty (first record), then return 20 milliseconds
+     */
+    public double getLastControlledTime() {
+        if (lastControlledTime.isEmpty()) {
+            return Constants.PERIOD_TIME_MILLISECONDS;
```

Comment:
- I think if this is empty, then you want to use this period as the difference between timestamps, not as the start timestamp
- `IntakeCal.INTAKE_PIVOT_FEEDFORWARD.calculate` is going to expect seconds, so either change this variable to seconds (making it 0.02) or divide it by 1000 where you do use it

Reply:
- updated the method to return the timestamp difference, returning 20 milliseconds if lastControlledTime is empty
- changed variable to seconds